sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -357,3 +360,35 @@ private[sql] object SortValue {
       None
   }
 }
+
+private[sql] sealed trait TimeTravelSpec
+
+private[sql] case class AsOfTimestamp(timestamp: Long) extends TimeTravelSpec
+private[sql] case class AsOfVersion(version: String) extends TimeTravelSpec
+
+private[sql] object TimeTravelSpec {
+  def create(timestamp: Option[String], version: Option[String]) : Option[TimeTravelSpec] = {
+    if (timestamp.nonEmpty && version.nonEmpty) {
+      throw new IllegalArgumentException(
+        "Version and Timestamp can't both be set in TimeTravelSpec")
```

Comment:
let's be more user-facing. `TimeTravelSpec` is an internal thing.

```
Cannot specify both version and timestamp when scanning the table.
```

Let's move it to `QueryCompilationErrors` as it's user-facing

Reply:
Fixed